### PR TITLE
fix: Permission state handling for notifications and media

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: app/script/calling/entities/CallEntity.js [hunk 1]
```diff
@@ -63,13 +63,7 @@ z.calling.entities.CallEntity = class CallEntity {
...
     this.callLogger = new z.telemetry.calling.CallLogger(loggerName, this.id, z.config.LOGGER.OPTIONS, this.messageLog);
 
-    this.callLogger.info({
-      data: {
-        default: [this.id],
-        obfuscated: [this.callLogger.obfuscate(this.id)],
-      },
-      message: `Created new call entity in conversation {0}`,
-    });
+    this.callLogger.info(`Created new call entity in conversation ${this.id}`);
```
Comment (atomrc) [inline app/script/calling/entities/CallEntity.js:66]: Seems we have lost obfuscation here

File: app/script/notification/NotificationRepository.js [hunk 2]
```diff
@@ -605,6 +603,15 @@ z.notification.NotificationRepository = class NotificationRepository {
     }
   }
 
+  /**
+   * Get the current browser permission state.
+   * @private
+   * @returns {Promise} Resolves with true if notifications are permitted
+   */
+  _updateBrowserPermissionState() {
```
Comment (atomrc) [inline app/script/notification/NotificationRepository.js]: Do we really need to extract this in another function?

File: app/script/media/MediaStreamHandler.js [hunk 3]
```diff
@@ -345,25 +345,31 @@ z.media.MediaStreamHandler = class MediaStreamHandler {
    *
    * @private
    * @param {z.media.MediaType} mediaType - Requested media type
-   * @returns {Promise} Resolves when the permissions are not denied
+   * @returns {Promise} Resolves true when permissions need to be prompted for
```
Comment (atomrc) [inline app/script/media/MediaStreamHandler.js]: The API of this method is confusing.
I would suggest to rename it to something like `hasPermissionToAccess(mediaType)` and reverse the returned boolean values

File: app/script/calling/entities/CallEntity.js (hunk 1 above)
Comment (gregor) [reply app/script/calling/entities/CallEntity.js:66]: According to @pwnsdx obfuscation of IP addresses and UUID is handled by default. We only need to pass in the object in cases where we replace names with IDs. Not exactly obvious. We need to see how to improve the documentation and handling of the call logger.

**Pull request merged**